src/qt/clientmodel.cpp

```diff
@@ -99,40 +98,21 @@ size_t ClientModel::getMempoolDynamicUsage() const
     return mempool.DynamicMemoryUsage();
 }
 
-double ClientModel::getVerificationProgress() const
+double ClientModel::getVerificationProgress(const CBlockIndex *tipIn) const
 {
-    LOCK(cs_main);
-    return Checkpoints::GuessVerificationProgress(Params().Checkpoints(), chainActive.Tip());
+    CBlockIndex *tip = (CBlockIndex *)tipIn;
```

Comment:
Better like this (it's now `const_cast<CBlockIndex *>(tipIn)`)?
Or would it make more sense to pass a non-const through the core signal?